### ISSUE TITLE
Rules function_body_length and type_body_length report incorrect number of lines

### ISSUE THREAD

**Issue by ornithocoder** [realm/SwiftLint#1750]
I'm working on a rule to limit the number of lines in a closure (#1747) and found an interesting issue with lines that have no characters other than curly braces: "{", "}".

When using the method `File.exceedsLineCountExcludingCommentsAndWhitespace(_:_:_:)`, `File.getter:syntaxKindsByLines` doesn't consider as valid lines those containing a single "{" or "}" character.

For instance, the code below:

```swift
foo.bar {
	let a = 0
	let a = 0
        // ...
	let a = 0
	if true {
		let a = 0
	}
}
```

returns the following `syntaxKindsByLines`. Note the last two lines, with 0 elements each.

```
▿ 26 elements
  ▿ 0 : 2 elements
    - 0 : SourceKittenFramework.SyntaxKind.identifier
    - 1 : SourceKittenFramework.SyntaxKind.identifier
  ▿ 1 : 3 elements
    - 0 : SourceKittenFramework.SyntaxKind.keyword
    - 1 : SourceKittenFramework.SyntaxKind.identifier
    - 2 : SourceKittenFramework.SyntaxKind.number
  ▿ 2 : 3 elements
    - 0 : SourceKittenFramework.SyntaxKind.keyword
    - 1 : SourceKittenFramework.SyntaxKind.identifier
    - 2 : SourceKittenFramework.SyntaxKind.number
  // ...
  ▿ 21 : 3 elements
    - 0 : SourceKittenFramework.SyntaxKind.keyword
    - 1 : SourceKittenFramework.SyntaxKind.identifier
    - 2 : SourceKittenFramework.SyntaxKind.number
  ▿ 22 : 2 elements
    - 0 : SourceKittenFramework.SyntaxKind.keyword
    - 1 : SourceKittenFramework.SyntaxKind.keyword
  ▿ 23 : 3 elements
    - 0 : SourceKittenFramework.SyntaxKind.keyword
    - 1 : SourceKittenFramework.SyntaxKind.identifier
    - 2 : SourceKittenFramework.SyntaxKind.number
  - 24 : 0 elements
  - 25 : 0 elements
```

Is this expected? I'd expect lines with "{" and "}" to return something, since they're valid lines.

**Comment by ornithocoder** [comment]
Another example:

```swift
func foo(a: Int) -> Int {
    if a < 0 {
        print("< 0")
    }

    return a
}
```

```swift
func testNumberOfLines() {
    let content = "func foo(a: Int) -> Int {\n" +
        "\tif a < 0 {\n" +
        "\t\tprint(\"< 0\")\n" +
        "\t}\n" +
        "\n" +
        "\treturn a\n" +
    "}"

    let file = File(contents: content)
    let emptyOrCommentCount = file.numberOfCommentAndWhitespaceOnlyLines(startLine: 1, endLine: 7)

    XCTAssertEqual(emptyOrCommentCount, 1) // returns 3
}
```

**Comment by marcelofabri** [comment]
@ornithocoder I haven't looked into this deeply, but IIRC the implementation, this would happen, and I'd consider it a bug. Feel free to open a PR improving `file.numberOfCommentAndWhitespaceOnlyLines`. 

It'd be just a matter of changing to the logic to "kindsOnLine.isEmpty && line.contents.trimmed.isEmpty" (pseudocode, not on my computer right now).

**Comment by ornithocoder** [comment]
These are the rules affected by this problem:

* `function_body_length`
* `type_body_length`
* `closure_body_length` (PR: #1747 for issue: #52)
